pkg/function/create_csi_snapshot.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	kanister "github.com/kanisterio/kanister/pkg"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/kube/snapshot"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+func init() {
+	_ = kanister.Register(&createCSISnapshotFunc{})
+}
+
+var (
+	_ kanister.Func = (*createCSISnapshotFunc)(nil)
+)
+
+const (
+	// CreateCSIVolumeSnapshotFuncName gives the name of the function
+	CreateCSISnapshotFuncName = "CreateCSISnapshot"
+	// CreateCSISnapshotNameArg provides name of the VolumeSnapshot to be created
```

Comment:
I think we should improve these comments, its the name of the volumesnapshot that should be created but we should also try to mention its argument to the function?